collector/cluster_health.go

```diff
@@ -0,0 +1,246 @@
+			"Whether all primary and replica shards are allocated.",
+			[]string{"cluster"}, nil,
+		),
+		StatusIsRed: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "status_is_red"),
+			"Whether all primary and replica shards are allocated.",
+			[]string{"cluster"}, nil,
+		),
+		TimedOut: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "timed_out"),
+			"XXX WHAT DOES THIS MEAN?",
+			[]string{"cluster"}, nil,
+		),
+		UnassignedShards: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, subsystem, "unassigned_shards"),
+			"The number of shards that exist in the cluster state, but cannot be found in the cluster itself.",
+			[]string{"cluster"}, nil,
+		),
+	}
+}
+
+func (c *ClusterHealth) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.Status
+}
+
+func (c *ClusterHealth) Collect(ch chan<- prometheus.Metric) {
+	c.url.Path = "/_cluster/health"
+	res, err := c.client.Get(c.url.String())
+	if err != nil {
+		level.Warn(c.logger).Log(
+			"msg", "failed to get cluster health",
+			"url", c.url.String(),
+			"err", err,
+		)
+		return
+	}
+	defer res.Body.Close()
+
+	dec := json.NewDecoder(res.Body)
+
+	var clusterHealthResponse clusterHealthResponse
+	if err := dec.Decode(&clusterHealthResponse); err != nil {
+		level.Warn(c.logger).Log(
+			"msg", "failed to decode cluster health",
+			"err", err,
+		)
+		return
+	}
+
+	cluster := clusterHealthResponse.ClusterName
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ActivePrimaryShards,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.ActivePrimaryShards),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.ActiveShards,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.ActiveShards),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.DelayedUnassignedShards,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.DelayedUnassignedShards),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.InitializingShards,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.InitializingShards),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.NumberOfDataNodes,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.NumberOfDataNodes),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.NumberOfInFlightFetch,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.NumberOfInFlightFetch),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.NumberOfNodes,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.NumberOfNodes),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.NumberOfPendingTasks,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.NumberOfPendingTasks),
+		cluster,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.RelocatingShards,
+		prometheus.GaugeValue,
+		float64(clusterHealthResponse.RelocatingShards),
+		cluster,
+	)
+
+	var statusValue float64
+	if clusterHealthResponse.Status == "green" {
```

Comment:
I currently assume that `elasticsearch_cluster_health_status` is `0` if red or yellow and only `1` when green.
I can only think about doing something like 
* `red` => `0`
* `yellow` => `1`
* `green` => `2` 
which honestly feels weird.